api/controller/EducationController.py

```diff
@@ -1,105 +1,128 @@
 
-        json_education_objects = {}
+            json_education_objects = {}
 
-        for education in all_education:
-            json_education_objects[convert_snake(education.classification)] = {
-                "type": str(education),
-                "id": str(education.id),
-                "attributes": {
-                    "classification": education.classification,
-                    "question": education.question,
-                    "description": education.description,
-                    "information": education.information,
-                    "note": education.note,
-                    "source": education.source
+            for education in all_education:
+                json_education_objects[convert_snake(education.classification)] = {
+                    "type": str(education),
+                    "id": str(education.id),
+                    "attributes": {
+                        "classification": education.classification,
+                        "question": education.question,
+                        "description": education.description,
+                        "information": education.information,
+                        "note": education.note,
+                        "source": education.source
+                    }
                 }
-            }
 
-        return {"data": json_education_objects}, 200
+            return {"data": json_education_objects}, 200
+        except Exception:
+            raise 500
 
     # POST
     def add_education(self):
-        # try:
-        body = request.get_json()
-        education = Education(**body)
-        education.save()
-        id = education.id
-        return {
-            "data": {
-                "id": str(id),
-                "confirmation": {
-                    "info": 'To see this new record, please do a GET request using the url',
-                    "url": f'/api/v1/education/{id}'
+        try:
+            body = request.get_json()
+            education = Education(**body)
+            education.save()
+            id = education.id
+            return {
+                "data": {
+                    "id": str(id),
+                    "confirmation": {
+                        "info": 'To see this new record, please do a GET request using the url',
+                        "url": f'/api/v1/education/{id}'
+                    }
                 }
-            }
-        }, 200
-        # except NotUniqueError():
-        #     raise EducationAlreadyExistsError
-        # except Exception:
-        #     raise InternalServerError
+            }, 200
+        except NotUniqueError:
+            raise APINotUniqueError("This education record's classification already exists in the database")
+        except (FieldDoesNotExist, ValidationError):
+            raise APISchemaError("Please check the Education documentation. Request is missing a required field or incorrect field entered.")
+        except Exception:
+            raise 500
```

Comment:
This is so cool!  I would love to get a walk through on this logic because it is both exciting and overwhelming!